common/guestchain/src/manager.rs

```diff
@@ -106,6 +107,18 @@ pub enum AddSignatureError {
     BadValidator,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum UpdateConfigError {
+    /// Minimum validators are more than existing
+    MinValidatorsHigherThanExisting,
+    /// Maximum validators are less than existing
+    MaxValidatorsLowerThanExisting,
+    /// Total Stake is less than existing
+    MinTotalStakeLowerThanExisting,
+    /// Quorum Stake is less than existing
+    MinQuorumStakeLowerThanExisting,
```

Comment:
for example, if the existing validators are 6 and we increase minimum validators to 7, then it doesnt make sense right?

Reply:
OK, but then the code doesn’t change the current number of validators but rather current configured minimum.  So if minimum is currently 5 but we have 10 validators it’s fine to increase the number to 7.

As to whether it makes sense, in general it is possible that chain is in such invalid state. This can happen when genesis block doesn’t match the configuration. The chain will work fine even if there’s fewer validators than minmum (it’s just that no one will be able to leave). I guess we can check for the condition but then we need to look at actual number of validators and also same with stake.

Reply:
oh yes right right.